src/Chapter1.hs

```diff
@@ -553,8 +554,14 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x < y && x < z && z < y = z
+    | x < y && x < z && y < z = y
+    | z < x && z < y && x < y = x
+    | z < x && z < y && y < x = y
+    | y < x && y < z && x < z = x
+    | otherwise = z
```

Comment:
Awesomely done!

One note: you can remove the first case here as it can be covered by the `otherwise` clause in this case:

```suggestion
    | x < y && x < z && y < z = y
    | z < x && z < y && x < y = x
    | z < x && z < y && y < x = y
    | y < x && y < z && x < z = x
    | otherwise = z
```

Reply:
ow --- that´s true - thanks